manager/controlapi/service.go

```diff
@@ -843,8 +843,28 @@ func (s *Server) RemoveService(ctx context.Context, request *api.RemoveServiceRe
 		return nil, status.Errorf(codes.InvalidArgument, errInvalidArgument.Error())
 	}
 
+	var service *api.Service
 	err := s.store.Update(func(tx store.Tx) error {
-		return store.DeleteService(tx, request.ServiceID)
+		service = store.GetService(tx, request.ServiceID)
+		if service == nil {
+			return status.Errorf(codes.NotFound, "service %s not found", request.ServiceID)
+		}
+
+		// if the service has no running tasks, we can remove it right away
+		m := service.Spec.GetMode()
+		switch m.(type) {
+		case *api.ServiceSpec_Replicated:
+			if int64(m.(*api.ServiceSpec_Replicated).Replicated.Replicas) == 0 {
```

Comment:
Replicated.Replicas is the replication target. I think there can be a case when the target is 0 but the actual tasks have not exited ?